modules/authentication/con-understanding-authentication-and-user-provisioning.adoc

```diff
@@ -0,0 +1,38 @@
+= Understanding authentication and user provisioning
+
+This module provides an overview of how authentication and user provisioning function within {product}.
+Learn about the process from user sign-in to the creation of user and group entities in the software catalog, and understand the roles played by the different authentication and catalog plugins.
+Understanding this process is essential for successfully link:{configuring-book-url}[configuring your {product-short} instance], link:{authorization-book-url}[securing access through authorization], and enabling features that rely on synchronized user and group data.
+
+When a user attempts to access {product-short}, {product-short} redirects them to a configured authentication provider, such as xref:assembly-authenticating-with-rhbk[{rhbk-brand-name} ({rhbk})], xref:authenticating-with-github[GitHub], or xref:assembly-authenticating-with-microsoft-azure[{azure-brand-name}].
+This external Identity Provider (IdP) is responsible for authenticating the user.
+
+On successful authentication, the {product-short}'s authentication plugin, configured in your `{my-app-config-file}` file, processes the response from the IdP, resolves the identity in the {product-short} software catalog, and establishes a user session within {product-short}.
+
+User and group data needs to be provisioned from the Identity Provider to the {product-short} software catalog to fully enable catalog features.
```

Comment:
I wonder if we are writing this in the order of user experience flow or set up flow for the platform engineer? If the latter, it might make sense to mention user provisioning first since it is a prerequisite to auth.

Reply:
*This* is exactly the kind of super useful input I always dream about.